python/cudf/cudf/dataframe/dataframe.py

```diff
@@ -1141,6 +1141,15 @@ def _sanitize_values(self, series, SCALAR):
            being added
         """
         if SCALAR:
+            if hasattr(series, "dtype") and (
+                series.dtype
+                in (
+                    np.dtype("datetime64[D]"),
+                    np.dtype("datetime64[M]"),
+                    np.dtype("datetime64[Y]"),
+                )
+            ):
```

Comment:
Instead of doing this should we just use `np.issubdtype(series.dtype, np.datetime64)`?

Reply:
Yes, changed to subdtype check

`= 11558 passed, 759 skipped, 85 xfailed, 6 xpassed, 1903 warnings in 377.46 seconds =`